### ISSUE TITLE
Also return the title of a vocabulary term in GET?

### ISSUE THREAD

**Issue by buchi** [plone/plone.restapi#726]
With the upcoming release we will return the token of a vocabulary term in GET requests. However for display purposes it would make sense to also return the title. Without the title it's necessary to get the required vocabularies and lookup up the title there.

As we are breaking the representation of vocabulary terms anyway, this would be a chance to go a step further and return an object containing the term and the title instead of just the token.

Example:

Current representation:
```json
{
  "language": "de"
}
```

Representation with token and title:
```json
{
  "language": {
    "title": "Deutsch",
    "token": "de"
  },
}
```

Or make this customizable with a query string parameter?
Related: #692

@tisto @sneridagh any thoughts?

**Comment by sneridagh** [comment]
@buchi We already take that into account and added a way to retrieve the title if required, given a token. This is what I implemented after our conversations in Sorrento. Take a look if it is what do you need. In Volto I already use it in some edge cases and works well. I still didn't find any case that I need it both at the same time, although it depends on how the vocabulary is built. 

https://github.com/plone/plone.restapi/blob/master/docs/source/vocabularies.rst#filter-vocabularies

**Comment by sneridagh** [comment]
Just tested the use case that you sent and the new vocabularies endpoint already returns title and token:

![image](https://user-images.githubusercontent.com/486927/57178735-50f56800-6e75-11e9-9588-f98a887929bd.png)

**Comment by sneridagh** [comment]
@buchi all is ready for release 4.0.0, but we didn't have time to release it. Did you test against master? I think that the key is to use your PR and the new vocabularies one.

**Comment by buchi** [comment]
@sneridagh My question/suggestion does not affect the vocabularies endpoint. I'm talking about the GET representation of a content item (e.g. a page).

If you fetch a content item, you only get the token. To display the title of the vocabulary term, you need to additionally fetch the vocabulary. Thus I would like to have a possibility to avoid that and get the title with the GET request of the content item.

**Comment by sneridagh** [comment]
@buchi oh, that is true... on display views, you'll be forced to do another request.

Then I'm +1 to return the title too along with the token. However, I'll do it before 4.0.0 since the changes are all related. I'll have to update Volto too accordingly too. When do you plan to implement the change?

**Comment by buchi** [comment]
I can implement that quickly. But there's still the question if we want to do that always or on demand by a querystring parameter (e.g. full_terms=1)?

**Comment by sneridagh** [comment]
It seems to me that have both in GETs, is valuable in any case... and it does not add any burden, right? At least at first thought... do you see any downside of returning both always?

**Comment by tisto** [comment]
I'd rather always return the title. We can always optimize later if we have to. 

FYI: I'd like to include plone.restapi 4.0.0 in the upcoming Plone 5.2 that @esteele is cutting right now if possible. Maybe we can try to get that fix in a release out of the door tomorrow?

**Comment by buchi** [comment]
Implemented in PR #727
